r/R/dplyr-funcs-datetime.R

```diff
@@ -105,7 +105,22 @@ register_bindings_datetime <- function() {
     (call_binding("yday", x) - 1) %/% 7 + 1
   })
 
-  register_binding("month", function(x, label = FALSE, abbr = TRUE, locale = Sys.getlocale("LC_TIME")) {
+  register_binding("month", function(x,
+                                     label = FALSE,
+                                     abbr = TRUE,
+                                     locale = Sys.getlocale("LC_TIME")) {
+
+    if (call_binding("is.integer", x)) {
+      if (is.integer(x)) {
+        x <- build_expr("cast", x, options = cast_options(to_type = int32()))
+      }
```

Comment:
This bit looks repetitive (these should already be integers — and this cast would then attempt to cast them to integers)? Are you trying to ensure that things aren't int64()? Or something else?

Reply:
It's an oversight from the initial structure which tried to handle things differently based on R object vs `Expression` logic.

Reply:
Removed duplicated code.